apps/desktop/src/autofill/modal/credentials/fido2-create.component.ts

```diff
@@ -94,34 +94,40 @@ export class Fido2CreateComponent implements OnInit, OnDestroy {
   ) {}
 
   async ngOnInit(): Promise<void> {
-    await this.accountService.setShowHeader(false);
     this.session = this.fido2UserInterfaceService.getCurrentSession();
     if (!this.session) {
       await this.showErrorDialog(DIALOG_MESSAGES.unableToSavePasskey);
       return;
     }
 
     try {
-      const relevantCiphers = await this.getRelevantCiphers();
-      this.ciphersSubject.next(relevantCiphers);
+      const displayedCiphers = await this.getDisplayedCiphers();
+      this.ciphersSubject.next(displayedCiphers);
     } catch {
       await this.showErrorDialog(DIALOG_MESSAGES.unexpectedErrorShort);
     }
   }
 
   async ngOnDestroy(): Promise<void> {
-    await this.accountService.setShowHeader(true);
+    await this.closeModal();
   }
 
-  async addPasskeyToCipher(cipher: CipherView): Promise<void> {
+  async addCredentialToCipher(cipher: CipherView): Promise<void> {
     const isConfirmed = await this.validateCipherAccess(cipher);
-    if (!this.session) {
+
+    try {
+      if (!this.session) {
+        await this.showErrorDialog(DIALOG_MESSAGES.unableToSavePasskey);
+        return;
+      }
+
+      this.session.notifyConfirmCreateCredential(isConfirmed, cipher);
+    } catch {
```

Comment:
suggestion: you could also just throw since you now have a try-catch
```suggestion

    try {
      if (!this.session) {
        throw new Error("Missing session");
      }

      this.session.notifyConfirmCreateCredential(isConfirmed, cipher);
    } catch {
```